### PR TITLE
Fe refactor payment

### PR DESCRIPTION
**Pull request by xia0m**
### What this PR does (required):
- refactor payment

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/6465981/121000830-84370f00-c73f-11eb-8b7a-55423ff120f6.mp4

### Any information needed to test this feature (required):
- use sitter@email.com, 123456 for dog sitter,
- use owner@email.com, 123456 for dog owner
- go to listing page, find Sitter Dog, and create a new request
- on sitter's panel, go to request page, the new request should be there
- click the vertical three dot to accept/decline the request
- if accept, owner is able to pay for the dog sitting.

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -46,6 +47,7 @@ function App(): JSX.Element {
                   <Route exact path="/listings" component={Listings} />
                   <Route exact path="/bookings" component={Bookings} />
                   <ProtectedRoute exact path="/user/:path" component={ProfileSettings} />
+                  <Route exact path="/requests" component={Request} />
```
Comment (rajivtitus) [inline client/src/App.tsx]: Please make this a protected route, although users who are not logged in won't be able to see any data but we wouldn't want them navigating to this page. While you are at it, could you also make bookings a protected route? :)

File: client/src/components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -60,18 +60,20 @@ export default function Navbar({ user, profile, logout }: Props): JSX.Element {
 
   const UserNav = () => (
     <Grid>
-      <Link to="/checkout" className={classes.link}>
-        <Button color="primary" className={`${classes.btn} ${classes.signupbtn}`} variant="contained">
-          Checkout
-        </Button>
-      </Link>
+      {profile?.isDogSitter ? (
+        <Link to="/requests" className={classes.link}>
```
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx]: You can merge Link and Button. 
Eg: `<Button component={Link} to='/requests'>Request</Button>`

File: client/src/components/ProfileDetails/RequestForm/RequestForm.tsx [hunk 3]
```diff
@@ -28,7 +27,6 @@ export default function RequestForm({ sitter }: Props): JSX.Element {
     if (sitter) {
       postRequest(sitter._id, startDate, endDate).then((data) => {
```
Comment (rajivtitus) [inline client/src/components/ProfileDetails/RequestForm/RequestForm.tsx]: Data is being defined but never used, please exclude 'data' and address warning.

File: client/src/pages/Checkout/Checkout.tsx [hunk 4]
```diff
@@ -5,36 +5,46 @@ import useStyles from './useStyles';
...
+import { Profile } from '../../interface/Profile';
+import { Request } from '../../interface/Bookings';
+
+import { useLocation } from 'react-router-dom';
+
+interface LocationState {
+  bookingDetails: Request;
+}
 
 export default function Order(): JSX.Element {
```
Comment (rajivtitus) [inline client/src/pages/Checkout/Checkout.tsx:17]: Once we are on Checkout page, please make Drop-In/Drop-Off dates and times read-only or maybe include the data in a Typography, as we wouldn't want users to change these details after sitter has confirmed the booking request.

Note: You can either work on this or create a new issue and assign it to one of us :)

File: client/src/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -60,18 +60,20 @@ export default function Navbar({ user, profile, logout }: Props): JSX.Element {
...
+          <Button variant="text" className={classes.userNavItem}>
+            Request
+          </Button>
+        </Link>
+      ) : null}
       <Button variant="text" className={classes.userNavItem}>
         Notifications <span className={classes.active} />
       </Button>
 
       <Link to="/bookings" className={classes.link}>
```
Comment (rajivtitus) [inline client/src/components/Navbar/Navbar.tsx]: Ditto here!

You can merge Link and Button.
Eg: `<Button component={Link} to='/requests'>Request</Button>`

**Review by rajivtitus — CHANGES_REQUESTED** [review]

**Comment by gursharan4312** [comment]
Couple of things:
- Change request id when it's passed to Payment component as it is hardcoded as of now
- You can add payment date and time in the request table when user has paid to confirm user that he has paid
- Please make a model or redirect to requests page when payment is successful.

**Review by gursharan4312 — APPROVED** [review]

**Pull request merged**